src/HDF5.jl

```diff
@@ -1928,17 +1933,26 @@ function __init__()
         ENV["HDF5_USE_FILE_LOCKING"] = "FALSE"
     end
 
+    __init_globals__()
     register_blosc()
 
+    # Generate the Float16 datatype:
```

Comment:
I would move this to `__init_globals()` and add some comments that this is added to support IEEE julia compatible Float16 values. We could also think about adding types of other types of Float16 dataypes that the Julia ecosystem supports.

Reply:
This has to happen after the `include("api.jl")` line which happens after this file.

Reply:
Perhaps then we can move the to a function elsewhere or in this file and then call that here. Something like `_init_custom_globals__()` ?

Reply:
Sure, I'd be OK with that.

Reply:
Oh wait, that was probably again caused by the temporary need to run `__init_globals__()` immediately during development. It should probably now work that actually running the function doesn't happen until `__init__`.